cmd/create/cluster/cmd.go

```diff
@@ -991,6 +991,79 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	cloudAccounts, err := r.OCMClient.GetBillingAccounts()
+	if err != nil {
+		r.Reporter.Errorf("%s", err)
+		os.Exit(1)
+	}
+
+	billingAccounts := ocm.GenerateBillingAccountsList(cloudAccounts)
+
+	if billingAccount == "" && !interactive.Enabled() {
+		// if a billing account is not provided we will try to use the infrastructure account as default
+		if helper.ContainsPrefix(billingAccounts, awsCreator.AccountID) {
+			billingAccount = awsCreator.AccountID
+			r.Reporter.Infof("Using '%s' as billing account", billingAccount)
+			r.Reporter.Infof("To use a different billing account, add --billing-account xxxxxxxxxx to previous command")
+		}
+	}
+
+	if interactive.Enabled() {
+		if len(billingAccounts) > 0 {
+			billingAccount, err = interactive.GetOption(interactive.Input{
+				Question: "Billing Account",
+				Help:     cmd.Flags().Lookup("billing-account").Usage,
+				Default:  billingAccount,
+				Required: true,
+				Options:  billingAccounts,
+			})
+
+			if err != nil {
+				r.Reporter.Errorf("Expected a valid billing account: '%s'", err)
+				os.Exit(1)
+			}
+
+			billingAccount = ocm.RegexpReturnOnlyInteger(billingAccount)
+		}
+
+		if billingAccount == "" || !ocm.IsValidAWSAccount(billingAccount) {
+			r.Reporter.Errorf("Expected a valid billing account")
+			os.Exit(1)
+		}
+
+		// Get contract info
+		var contracts []*accountsv1.Contract
+		isContractEnabled := false
+		for _, account := range cloudAccounts {
+			if account.CloudAccountID() == billingAccount {
+				contracts = account.Contracts()
+				for _, contract := range contracts {
+					for _, dimension := range contract.Dimensions() {
+						if ocm.HasValidContracts(dimension) {
+							isContractEnabled = true
+							break
+						}
+					}
+				}
+
+			}
+		}
+
+		if billingAccount != awsCreator.AccountID {
+			r.Reporter.Infof("The selected AWS billing account is a different account than your AWS infrastructure account." +
+				"The AWS billing account will be charged for subscription usage. " +
+				"The AWS infrastructure account will be used for managing the cluster. \n")
+		} else {
+			r.Reporter.Infof("Using '%s' as billing account.",
+				billingAccount)
+		}
+
+		if isContractEnabled {
+			PrintContracts(contracts, r)
+		}
```

Comment:
This will print contracts for all accounts - you probably want to print the contracts for the selected account

Reply:
It's getting the contracts from the selected billing account: https://github.com/openshift/rosa/pull/1500/files#diff-943c0ab57b82c7028652df55e1d49e4f4e4dc41fa094dc4a9b3f2c35887d44b5R1039